tikv-client/src/main/java/com/pingcap/tikv/expression/aggregate/Sum.java

```diff
@@ -20,6 +21,12 @@ protected ExprType getExprType() {
 
   @Override
   public DataType getType() {
-    return DataTypeFactory.of(Types.TYPE_NEW_DECIMAL);
+    // get column type from Column Reference
+    DataType colType = args.get(0).getType();
+    if(colType instanceof DecimalType) {
+      return DataTypeFactory.of(Types.TYPE_NEW_DECIMAL);
+    }
+    // both Real and Double are decoded using RealType
+    return DataTypeFactory.of(Types.TYPE_DOUBLE);
```

Comment:
This logic is not correct since `sum(int)` will be of NEW_DECIMAL type.

Reply:
The correct logic may probably look like this: 
```
public DataType getType() {
   DataType colType = args.get(0).getType();
   if(colType instanceof RealType) {
     return DataTypeFactory.of(Types.TYPE_DOUBLE);
   }
   return DataTypeFactory.of(Types.TYPE_NEW_DECIMAL);
}
```

Reply:
There is no difference between you and mine. Remember that, sum only accept `int`, `float`, and `decimal `.